src/ergw_aaa_radius.erl

```diff
@@ -358,6 +358,9 @@ session_options('Acct-Authentic', 'Remote', Attrs) ->
 session_options('Acct-Authentic', 'Diameter', Attrs) ->
     [{?Acct_Authentic, 4}|Attrs];
 
+session_options('Termination-Cause', Value, Attrs) ->
```

Comment:
No. We use DIAMETER code when passing the TermCause. RADIUS values are different. A mapping is needed here.

One possibility would be to use atoms for it everywhere and then map to the corresponding RADIUS and DIAMETER codes. This would also need some larger (but straight forward changes in ergw).

Reply:
DIAMETER: https://www.iana.org/assignments/aaa-parameters/aaa-parameters.xhtml#aaa-parameters-16
RADIUS: https://www.iana.org/assignments/radius-types/radius-types.xhtml#radius-types-12

Reply:
Thanks ! The cause is composed at only few places in ergw e.g. : https://github.com/travelping/ergw/blob/master/src/ggsn_gn.erl#L598 . Would it be an idea to expose the Reason directly in the session to AAA (e.g. as Termination-Int-Reason) ?

Reply:
You mean like including two different term causes? That sounds like a ugly hack.

Reply:
No, only the internal reason instead of the current Termination-Cause ...